From: David Woodhouse <dwmw@amazon.co.uk>
Date: Sat, 24 Oct 2020 22:35:33 +0100
Subject: iommu/hyper-v: Disable IRQ pseudo-remapping if 15 bit APIC IDs are available
Git-commit: bf27ef8a77d8da38c9f35f8f6aab013a2dcf175f
Patch-mainline: v5.11-rc1
References: jsc#SLE-16823

If the 15-bit APIC ID support is present in emulated MSI then there's no
need for the pseudo-remapping support.

Signed-off-by: David Woodhouse <dwmw@amazon.co.uk>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Link: https://lore.kernel.org/r/20201024213535.443185-34-dwmw2@infradead.org
Acked-by: Joerg Roedel <jroedel@suse.de>
---
 drivers/iommu/hyperv-iommu.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/iommu/hyperv-iommu.c
+++ b/drivers/iommu/hyperv-iommu.c
@@ -121,6 +121,7 @@ static int __init hyperv_prepare_irq_rem
 	int i;
 
 	if (!hypervisor_is_type(X86_HYPER_MS_HYPERV) ||
+	    x86_init.hyper.msi_ext_dest_id() ||
 	    !x2apic_supported())
 		return -ENODEV;
 
